app/helpers/application_helper.rb

```diff
@@ -73,24 +73,61 @@ def alert_container_class_for_flash_type type
 
   def filter_badge(label: nil, model: nil, filter_by: nil, color: nil, title: nil)
     if model.present?
-      url = "/#{model}?#{filter_by}=#{CGI.escape(label)}"
+      query_string = build_query_string(toggle_filter(filter_by, label))
+      url = "/#{model}?#{query_string}"
     end
 
+    applied = get_query_params[filter_by].include? label
     case color
-    when 'gray'
-      classes = 'bg-gray-100 text-gray-800'
-      classes += ' bg-gray-200' if request.fullpath == url
     when 'blue'
       classes = 'bg-blue-100 text-blue-800'
-      classes += ' bg-blue-200' if request.fullpath == url
+      classes += ' bg-blue-200' if applied
     else
       classes = 'bg-indigo-100 text-indigo-800'
-      classes += ' bg-indigo-200' if request.fullpath == url
+      classes += ' bg-indigo-200' if applied
```

Comment:
I would go with a darker shade for better contrast.

```suggestion
      classes += ' bg-indigo-300' if applied
```